### PR TITLE
time: add accurate sleep procedure

### PR DESCRIPTION
**Pull request by ftphikari**
This is initial implementation, needs some work and testing to do.

### CODE REVIEW

File: core/time/time.odin [hunk 1]
```diff
@@ -213,6 +213,43 @@ time_add :: proc(t: Time, d: Duration) -> Time {
 	return Time{t._nsec + i64(d)}
 }
 
+yield :: proc "contextless" () {
```
Comment (Kelimion) [inline core/time/time.odin:216]: This is unnecessary, just call `_yield()`.

**Comment by ftphikari** [comment]
Here are some results on Windows 10 core2duo:

```
errors for 1 milliseconds:
...........
sleep errors:
	minimal: 1.017247ms
	latest: 1.034276ms
	average: 1.03253ms
...........
accurate sleep errors:
	minimal: 3.282µs
	latest: 3.338µs
	average: 19.772µs

errors for 2 milliseconds:
...........
sleep errors:
	minimal: 235.354µs
	latest: 259.285µs
	average: 247.109µs
...........
accurate sleep errors:
	minimal: 12.714µs
	latest: 12.721µs
	average: 13.131µs

errors for 3 milliseconds:
...........
sleep errors:
	minimal: 337.804µs
	latest: 363.896µs
	average: 375.145µs
...........
accurate sleep errors:
	minimal: 24.962µs
	latest: 25.112µs
	average: 27.417µs

errors for 4 milliseconds:
...........
sleep errors:
	minimal: 453.819µs
	latest: 558.15µs
	average: 568.134µs
...........
accurate sleep errors:
	minimal: 73.137µs
	latest: 73.4µs
	average: 78.031µs

errors for 5 milliseconds:
...........
sleep errors:
	minimal: 623.742µs
	latest: 798.152µs
	average: 677.797µs
...........
accurate sleep errors:
	minimal: 136.285µs
	latest: 212.297µs
	average: 144.064µs
```

**Comment by ftphikari** [comment]
This is the program that was used:
```odin
package sleep_test

import "core:time"
import "core:fmt"

test_sleep_proc :: proc(sleep_proc: proc(time.Duration), interval: time.Duration) -> (minimal, maximal, latest, average: time.Duration) {
	tests :: 10

	start: time.Tick
	end: f64
	avg: time.Duration
	avg_count: int

	runs := 1000 / time.duration_milliseconds(interval)
	expected := time.duration_microseconds(interval) * f64(runs)

	minimal = -1

	for in 0..tests {
		start = time.tick_now()
		for in 0..runs {
			sleep_proc(interval)
		}
		end = time.duration_microseconds(time.tick_since(start))
		error := time.Duration(abs(end - expected) * f64(time.Microsecond) / f64(cast(f64)runs / time.duration_milliseconds(interval)))
		latest = error
		if minimal < 0 do minimal = error
		minimal = min(minimal, error)
		maximal = max(maximal, error)
		avg += error
		avg_count += 1
		fmt.print(".")
	}
	fmt.println()

	return minimal, maximal, latest, time.Duration(f64(avg)/f64(avg_count))
}

main :: proc() {
	minimal, maximal, latest, average: time.Duration
	for i in 1..5 {
		interval := time.Duration(i) * time.Millisecond

		fmt.println("errors for", i, "milliseconds:")

		minimal, maximal, latest, average = test_sleep_proc(time.sleep, interval)

		fmt.println("sleep errors:")
		fmt.println("	minimal:", minimal)
		fmt.println("	latest:", latest)
		fmt.println("	average:", average)

		minimal, maximal, latest, average = test_sleep_proc(time.accurate_sleep, interval)

		fmt.println("accurate sleep errors:")
		fmt.println("	minimal:", minimal)
		fmt.println("	latest:", latest)
		fmt.println("	average:", average)
		fmt.println()
	}
}
```

**Comment by ftphikari** [comment]
Note: after doing `win32.timeBeginPeriod(1)`  the timings become precise immediately, that's because Windows scheduler takes time to adjust to your program. Here are the results after asking the scheduler to lower the interval:
```
errors for 1 milliseconds:
...
sleep errors:
	minimal: 996.723µs
	maximal: 1.013619ms
	latest: 1.013619ms
	average: 1.005421ms
...
accurate sleep errors:
	minimal: 2.532µs
	maximal: 13.168µs
	latest: 2.942µs
	average: 6.214µs

errors for 2 milliseconds:
...
sleep errors:
	minimal: 654.178µs
	maximal: 717.308µs
	latest: 696.801µs
	average: 689.429µs
...
accurate sleep errors:
	minimal: 5.548µs
	maximal: 28.32µs
	latest: 28.32µs
	average: 13.465µs

errors for 3 milliseconds:
...
sleep errors:
	minimal: 848.667µs
	maximal: 984.192µs
	latest: 984.192µs
	average: 927.824µs
...
accurate sleep errors:
	minimal: 26.769µs
	maximal: 36.268µs
	latest: 27.431µs
	average: 30.156µs

errors for 4 milliseconds:
...
sleep errors:
	minimal: 919.305µs
	maximal: 1.090155ms
	latest: 982.849µs
	average: 997.436µs
...
accurate sleep errors:
	minimal: 9.577µs
	maximal: 27.436µs
	latest: 9.577µs
	average: 15.869µs

errors for 5 milliseconds:
...
sleep errors:
	minimal: 1.046727ms
	maximal: 1.244052ms
	latest: 1.046727ms
	average: 1.140847ms
...
accurate sleep errors:
	minimal: 11.965µs
	maximal: 41.292µs
	latest: 11.965µs
	average: 21.955µs
```

**Pull request merged**